lib/rules/function-paren-newline.js

```diff
@@ -191,10 +191,11 @@ module.exports = {
         function getParenTokens(node) {
             switch (node.type) {
                 case "NewExpression":
-                    if (!node.arguments.length && !(
-                        astUtils.isOpeningParenToken(sourceCode.getLastToken(node, { skip: 1 })) &&
+                    if (node.callee.type === "NewExpression" ||
+                        !node.arguments.length && !(
+                            astUtils.isOpeningParenToken(sourceCode.getLastToken(node, { skip: 1 })) &&
                         astUtils.isClosingParenToken(sourceCode.getLastToken(node))
```

Comment:
This would fail to check parentheses of the outer `NewExpression` when it does have its own parentheses.

```js
/* eslint function-paren-newline: ["error", "never"] */

// false negative
new new C()(
);

```

Instead of checking if `callee` is another `NewExpression`, we could directly check if the parentheses are of the outer `NewExpression` by looking at ranges:

```js
if (!node.arguments.length && !(
    astUtils.isOpeningParenToken(sourceCode.getLastToken(node, { skip: 1 })) &&
    astUtils.isClosingParenToken(sourceCode.getLastToken(node)) &&
    node.callee.range[1] < node.range[1]
)) {

    // If the NewExpression does not have parens (e.g. `new Foo`), return null.
    return null;
}
```

Reply:
Good catch! Just updated.